lib/CXGN/Genotype/Delete.pm

```diff
@@ -0,0 +1,192 @@
+
+=head1 NAME
+
+CXGN::Genotype::Delete - an object to delete plate, project or protocol genotyping data
+
+=head1 DESCRIPTION
+
+    my $genotyping_data_delete = CXGN::Genotype::Delete->new( { schema => $schema, plate_id => 37347 });
+
+
+=head1 AUTHORS
+
+    Titima Tantikanjana
+
+=head1 METHODS
+
+=cut
+package CXGN::Genotype::Delete;
+
+use Moose;
+use SGN::Model::Cvterm;
+use Data::Dumper;
+use JSON;
+use Try::Tiny;
+use CXGN::Trial;
+use CXGN::Trial::TrialLayout;
+use CXGN::Stock::TissueSample::Search;
+
+has 'bcs_schema' => (
+    isa => 'Bio::Chado::Schema',
+    is => 'rw',
+    required => 1,
+);
+
+has 'protocol_id' => (
+    isa => 'Int',
+    is => 'rw',
+);
+
+has 'genotyping_project_id' => (
+    isa => 'Int',
+    is => 'rw',
+);
+
+has 'genotyping_plate_id' => (
+    isa => 'Int',
+    is => 'rw',
+);
+
+has 'empty_protocol_id' => (
+    isa => 'Int',
+    is => 'rw',
+);
+
+sub delete_genotype_data {
+    my $self = shift;
+    my $schema = $self->bcs_schema();
+    my $genotyping_project_id = $self->genotyping_project_id();
+    my $genotyping_plate_id = $self->genotyping_plate_id();
+    my $genotyping_protocol_id = $self->protocol_id();
+    my $dbh = $schema->storage->dbh;
+    my $empty_protocol_name;
+    my $empty_protocol_id;
+
+    eval {
+        $dbh->begin_work();
+
+        my $where_clause;
+        my $experiment_cvterm_id = SGN::Model::Cvterm->get_cvterm_row($schema, 'genotyping_experiment', 'experiment_type')->cvterm_id();
+
+        if ($genotyping_plate_id) {
+            my @plate_list = ();
+            @plate_list = ($genotyping_plate_id);
+            my $plate_samples = CXGN::Stock::TissueSample::Search->new({
+                bcs_schema => $schema,
+                plate_db_id_list => \@plate_list,
+            });
+
+            my $data = $plate_samples->get_sample_data();
+            my $sample_list = $data->{sample_list};
+            my $stock_ids = join ("," , @$sample_list);
+            $where_clause = "nd_experiment_stock.stock_id in ($stock_ids)";
+
+        } elsif ($genotyping_project_id) {
+            $where_clause = "nd_experiment_project.project_id = $genotyping_project_id";
+        } elsif ($genotyping_protocol_id) {
+            $where_clause = "nd_experiment_protocol.nd_protocol_id = $genotyping_protocol_id";
```

Comment:
We should try to use only placeholders in sql statements. Could you replace the embedded variables with placeholders (?) ?

Reply:
@lukasmueller I have been using placeholders in all cases, except this type of statements "join ("," , @$sample_list)", because it doesn't work with the placeholder.